sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -848,15 +856,25 @@ export class MessageReceiver extends LinkEntity {
 
   /**
    * Will reconnect the receiver link if necessary.
-   * @param {AmqpError | Error} [receiverError] The receiver error if any.
+   * @param receiverError The receiver error if any.
+   * @param connectionDidDisconnect Whether this method is called as a result of a connection disconnect.
    * @returns {Promise<void>} Promise<void>.
    */
-  async onDetached(receiverError?: AmqpError | Error): Promise<void> {
+  async onDetached(
+    receiverError?: AmqpError | Error,
+    connectionDidDisconnect?: boolean
```

Comment:
I added a comment above about using a string union here. Might be nice so then it's like "source" or something of that nature?

Reply:
Renamed the variable to be less confusing.

Reply:
Yes, yes you did. It's 100% better.